src/core/execute.c

```diff
@@ -3124,6 +3129,46 @@ static int compile_bind_mounts(
         return r;
 }
 
+static int compile_symlinks(
+                const ExecContext *context,
+                const ExecParameters *params,
+                char ***ret_symlinks) {
+
+        _cleanup_strv_free_ char **symlinks = NULL;
+        int r;
+
+        assert(context);
+        assert(params);
+        assert(ret_symlinks);
```

Comment:
maybe add a comment somewhere that explains what ret_symlinks contains, given it's not a plain list of things, but a list of pairs of which each one has a specific meaning.